docs/source/tutorial_chat_service.md

```diff
@@ -0,0 +1,152 @@
+- The `chat` task is an example of a task that requires multiple users, and shows how many people can be connected together in an instance of a world and then returned to the overworld upon completion of a task.
+
+In addition to the overworld demo, the following example tasks are provided:
+
+- [Generic Chatbot](https://github.com/facebookresearch/ParlAI/blob/master/parlai/chat_service/tasks/chatbot/): Allow messenger conversations with any ParlAI models, for instance the [PersonaChat](https://github.com/facebookresearch/ParlAI/tree/master/projects/personachat) model.
+- [QA Data Collection](https://github.com/facebookresearch/ParlAI/blob/master/parlai/chat_service/tasks/qa_data_collection/): collect questions and answers from people, given a random Wikipedia paragraph from SQuAD.
+
+### Creating your Own Task
+
+To create your own task, start with reading the tutorials on the provided examples, and then copy and modify the example `worlds.py` and `config.yml` files to create your task.
+
+**A few things to keep in mind:**
+
+1. A conversation ends when a call between `parley` calls to `episode_done` returns True.
+2. Tasks with an overworld should return the name of the world that they want to queue a user into from the ``parley`` call in which the user makes that selection to enter a world.
+3. Tasks with no overworld will immediately attempt to put a user into the queue for the default task onboarding world or actual task world (if no onboarding world exists), and will do so again following the completion of a world (via `episode_done`).
+3. To collect the conversation, data should be collected during every `parley` and saved during the `world.shutdown` call. You must inform the user of the fact that the data is being collected as well as your intended use.
+4. Finally, if you wish to use and command line arguments as you would in ParlAI, specify those in the `opt` section of the config file.
+
+
+## Available Chat Services
+
+
+### Browser
+This allows you to participate in a ParlAI world as an agent using a local browser.
+This extends the `websocket` chat service implementation to run a server locally,
+which you can send and receive messages using a browser.
+
+#### Setup
+
+1. Run: `python parlai/chat_service/services/browser_chat/run.py --config-path path/to/config.yml --port PORT_NUMBER`
+2. Run: `python client.py --port PORT_NUMBER`
+3. Interact
+
+**Example command:** `python parlai/chat_service/services/browser_chat/run.py --config-path parlai/chat_service/tasks/chatbot/config.yml --port 10001`
+
+If no port number is specified in `--port` then the default port used will be `34596`. If specifying, ensure both port numbers match on client and server side.
+
+
+### Facebook Messenger
+
+This allows you to chat with a ParlAI model on Facebook Messenger.
+
+<p align="center"><img width="90%" src="docs/source/_static/img/messenger-example.png" /></p>
```

Comment:
When I preview this it seems to go to `docs/source/docs/source/_static/img/messenger-example.png`. I don't think the first `docs/source` is required?

Reply:
thanks for pointing that out!